packages/fxa-auth-server/test/local/sqs.js

```diff
@@ -0,0 +1,58 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+const { assert } = require('chai');
+const sinon = require('sinon');
+
+const ROOT_DIR = '../..';
+let SQSReceiver, statsd, testQueue;
+const log = { error: sinon.stub() };
+
+describe('SQSReceiver', () => {
+  beforeEach(() => {
+    statsd = { timing: sinon.stub() };
+    SQSReceiver = require(`${ROOT_DIR}/lib/sqs`)(log, statsd);
+    testQueue = new SQSReceiver('testo', [
+      'https://sqs.testo.meows.xyz/fxa/quux',
+    ]);
+    const receiveStub = sinon.stub();
+    receiveStub.onFirstCall().callsFake((qParams, cb) => {
+      cb(null, { Messages: [JSON.stringify({ Body: 'SYN' })] });
+    });
+    receiveStub.returns(null);
```

Comment:
The callback is [recursive](https://github.com/mozilla/fxa/blob/28d25cfd6c26d9232c1e36310b7b29efc8763cac/packages/fxa-auth-server/lib/sqs.js#L87).  This is to prevent the recursion.